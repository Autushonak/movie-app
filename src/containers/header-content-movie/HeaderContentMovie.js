import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import searchIcon from '../../static/search-icon.jpg';
import { Title } from '../../components';
import { colors } from '../../constants/colors';

const HeaderContentMovieStyled = styled.div`
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  position: relative;
  width: 100%;
  height: ${({ height }) => height}px;
  z-index: 9999;
  padding: 100px 100px;
  box-sizing: border-box;
`;

const Poster = styled.img`
  height: 100%;
  width: 250px;
  margin-right: 70px;
`;

const MovieDescription = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
`;

const Row = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const RatingBox = styled.div`
  border: 1px solid ${colors.WHITE};
  padding: 10px 10px;
  border-radius: 50%;
  margin-left: 30px;
`;

const Rating = styled.h3`
  color: ${colors.GREEN};
  margin: 0;
`;

const Text = styled.p`
  color: ${colors.WHITE};
`;

const MovieInfo = styled.div`
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  color: ${colors.WHITE};
  margin-right: 30px;
`;

const MovieInfoProperty = styled.p`
  font-size: small;
  color: ${colors.GREY_0};
  margin: 0;
`;
const MovieInfoValue = styled.h3`
  color: ${colors.PINK};
  margin: 0 8px 0 0;
`;

const SearchIcon = styled.img`
  position: absolute;
  top: 20px;
  right: 20px;
  width: 40px;
  height: 40px;
`;

const movie = {
  title: 'Kill Bill Vol.2',
  imageURL: 'https://image.tmdb.org/t/p/w500/6qmsupE0opYPIaBGe7T5D2FBzLs.jpg',
  description:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
  oscarWinning: true,
  year: '1994',
  length: 154,
  rating: 4.3,
};

const HeaderContentMovie = ({ height }) => (
  <HeaderContentMovieStyled height={height}>
    <SearchIcon src={searchIcon}></SearchIcon>
    <Poster src={movie.imageURL}></Poster>
    <MovieDescription>
      <Row>
        <Title>{movie.title}</Title>
        <RatingBox>
          <Rating>{movie.rating}</Rating>
        </RatingBox>
      </Row>
      <Row>
        <Text>Oscar winning Movie</Text>
      </Row>
      <Row>
        <MovieInfo>
          <MovieInfoValue>{movie.year}</MovieInfoValue>
          <MovieInfoProperty>year</MovieInfoProperty>
        </MovieInfo>
        <MovieInfo>
          <MovieInfoValue>{movie.length}</MovieInfoValue>
          <MovieInfoProperty>min</MovieInfoProperty>
        </MovieInfo>
      </Row>
      <Row>
        <Text>{movie.description}</Text>
      </Row>
    </MovieDescription>
  </HeaderContentMovieStyled>
);

HeaderContentMovie.propTypes = {
  height: PropTypes.number,
};

export default HeaderContentMovie;
export {
  HeaderContentMovieStyled,
  Poster,
  MovieDescription,
  Row,
  RatingBox,
  Rating,
  Text,
  MovieInfo,
  MovieInfoProperty,
  MovieInfoValue,
  SearchIcon,
};
