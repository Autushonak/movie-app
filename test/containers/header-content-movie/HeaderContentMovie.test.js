import 'jest-styled-components';
import React from 'react';
import { shallow } from 'enzyme';
import HeaderContentMovie, {
  HeaderContentMovieStyled,
  Poster,
  MovieDescription,
  Row,
  RatingBox,
  Rating,
  Text,
  MovieInfo,
  MovieInfoProperty,
  MovieInfoValue,
  SearchIcon,
} from '../../../src/containers/header-content-movie/HeaderContentMovie';

describe('HeaderContentMovieStyled', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<HeaderContentMovieStyled height={500} />);
    expect(component).toMatchSnapshot();
  });
});

describe('Poster', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<Poster />);
    expect(component).toMatchSnapshot();
  });
});

describe('MovieDescription', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<MovieDescription />);
    expect(component).toMatchSnapshot();
  });
});

describe('Row', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<Row />);
    expect(component).toMatchSnapshot();
  });
});

describe('RatingBox', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<RatingBox />);
    expect(component).toMatchSnapshot();
  });
});

describe('Rating', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<Rating />);
    expect(component).toMatchSnapshot();
  });
});

describe('Text', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<Text />);
    expect(component).toMatchSnapshot();
  });
});

describe('MovieInfo', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<MovieInfo />);
    expect(component).toMatchSnapshot();
  });
});

describe('MovieInfoProperty', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<MovieInfoProperty />);
    expect(component).toMatchSnapshot();
  });
});

describe('MovieInfoValue', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<MovieInfoValue />);
    expect(component).toMatchSnapshot();
  });
});

describe('SearchIcon', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<SearchIcon />);
    expect(component).toMatchSnapshot();
  });
});

describe('HeaderContentMovie', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<HeaderContentMovie height={500} />);
    expect(component).toMatchSnapshot();
  });
});
