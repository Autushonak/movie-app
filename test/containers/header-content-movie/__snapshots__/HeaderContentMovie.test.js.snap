// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContentMovie should be rendered correctly 1`] = `
<styled.div
  height={500}
>
  <a
    href="/"
  >
    <styled.img
      src=""
    />
  </a>
  <styled.img
    src="test_path"
  />
  <styled.div>
    <styled.div>
      <styled.h1>
        test-title
      </styled.h1>
      <styled.div>
        <styled.h3>
          123
        </styled.h3>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.h3>
          2020
        </styled.h3>
        <styled.p>
          year
        </styled.p>
      </styled.div>
      <styled.div>
        <styled.h3>
          123
        </styled.h3>
        <styled.p>
          min
        </styled.p>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.p>
        test overview
      </styled.p>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`HeaderContentMovieStyled should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
  width: 100%;
  height: 500px;
  z-index: 9999;
  padding: 100px 100px;
  box-sizing: border-box;
}

<div
  className="c0"
  height={500}
/>
`;

exports[`MovieDescription should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;

exports[`MovieInfo should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: #FFFFFF;
  margin-right: 30px;
}

<div
  className="c0"
/>
`;

exports[`MovieInfoProperty should be rendered correctly 1`] = `
.c0 {
  font-size: small;
  color: #555555;
  margin: 0;
}

<p
  className="c0"
/>
`;

exports[`MovieInfoValue should be rendered correctly 1`] = `
.c0 {
  color: #F65261;
  margin: 0 8px 0 0;
}

<h3
  className="c0"
/>
`;

exports[`Poster should be rendered correctly 1`] = `
.c0 {
  height: 100%;
  width: 250px;
  margin-right: 70px;
}

<img
  className="c0"
/>
`;

exports[`Rating should be rendered correctly 1`] = `
.c0 {
  color: #A1E66F;
  float: left;
  clear: left;
  margin: 0;
  padding: 0;
}

<h3
  className="c0"
/>
`;

exports[`RatingBox should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 1px solid #FFFFFF;
  padding: 10px 10px;
  height: 20px;
  width: 20px;
  border-radius: 50%;
  margin-left: 30px;
}

<div
  className="c0"
/>
`;

exports[`Row should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`SearchIcon should be rendered correctly 1`] = `
.c0 {
  position: absolute;
  top: 20px;
  right: 20px;
  width: 40px;
  height: 40px;
}

<img
  className="c0"
/>
`;

exports[`Text should be rendered correctly 1`] = `
.c0 {
  color: #FFFFFF;
}

<p
  className="c0"
/>
`;
